desktop-src/direct3dhlsl_current/hlsl-method-sample.md

```diff
@@ -0,0 +1,226 @@
+
+Type: uint
+
+## Remarks
+
+Texture sampling uses the texel position to look up a texel value. An offset can be applied to the position before lookup. The sampler state contains the sampling and filtering options. This method can be invoked within a pixel shader, but it is not supported in a vertex shader or a geometry shader.
+
+Use an offset only at an integer miplevel; otherwise, you may get different results depending on hardware implementation or driver settings.
+
+### Calculating texel positions
+
+Texture coordinates are floating-point values that reference texture data, which is also known as normalized texture space. Address wrapping modes are applied in this order (texture coordinates + offsets + wrap mode) to modify texture coordinates outside the [0...1] range.
+
+For texture arrays, an additional value in the location parameter specifies an index into a texture array. This index is treated as a scaled float value (instead of the normalized space for standard texture coordinates). The conversion to an integer index is done in the following order (float + round-to-nearest-even integer + clamp to the array range).
+
+### Applying texture coordinate offsets
+
+The offset parameter modifies the texture coordinates, in texel space. Even though texture coordinates are normalized floating-point numbers, the offset applies an integer offset. Also note that the texture offsets need to be static.
+
+The data format returned is determined by the texture format. For example, if the texture resource was defined with the DXGI_FORMAT_A8B8G8R8_UNORM_SRGB format, the sampling operation converts sampled texels from gamma 2.0 to 1.0, filter, and writes the result as a floating-point value in the range [0..1].
+
+## Example
+
+```HLSL
+// Object Declarations
+Texture2D g_MeshTexture;            // Color texture for mesh
+SamplerState MeshTextureSampler;
+
+struct VS_OUTPUT
+{
+    float4 Position   : SV_POSITION; // vertex position
+    float4 Diffuse    : COLOR0;      // vertex diffuse color
+    float2 TextureUV  : TEXCOORD0;   // vertex texture coords
+};
+
+float4 PSMain(VS_OUTPUT In) : SV_Target
+{
+  return g_MeshTexture.Sample(MeshTextureSampler, In.TextureUV) * In.Diffuse;
+}
+```
+
+## Supported Shader Models
+
+| Shader Model | 6.0 | 6.1 | 6.2 | 6.3 | 6.4 | 6.5 | 6.6 |
+| --- | --- | --- | --- | --- | --- | --- | --- |
+| Vertex | - | - | - | - | - | - | - |
+| Hull | - | - | - | - | - | - | - |
+| Domain | - | - | - | - | - | - | - |
+| Geometry | - | - | - | - | - | - | - |
+| Amplification | - | - | - | - | - | - | 1* |
+| Mesh | - | - | - | - | - | - | 1* |
+| Pixel | x | x | x | x | x | x | x |
+| Compute | - | - | - | - | - | - | x |
+
+| Key | Description |
+| - | - |
+| x | Supported |
+| 1* | Supported with optional feature: `SHADER_FEATURE_DERIVATIVES_IN_MESH_AND_AMPLIFICATION_SHADERS` |
+
+>TBD: Make optional feature into a link
```

Comment:
This TBD should be resolved and removed from here.  It's replicated all over the place, along with the table that isn't correct for most other operations.

Reply:
I think we should remove these compatibility tables until we have a good method of researching compatibility.